### PR TITLE
ADEN-1586 Update screenshot comparison tests.

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java [hunk 1]
```diff
@@ -67,4 +69,62 @@ public boolean isColorImage(BufferedImage image, Color color, int accuracy) {
...
+        }
+      }
+      if (sameCount > ((100 - threshold) * count) / 100D) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public Triple getRgbVariance(BufferedImage image) {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java]: Add javadoc so in the future, everyone knows thar "middle" is green variance. Even better - make a class of image that will hold all the data for image - variance, count and make calculate the difference.

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageEditor.java [hunk 2]
```diff
@@ -34,9 +34,7 @@ public void saveImageFile(File imageFile, String path) {
     }
   }
 
-  public BufferedImage scaleImage(
-      File inputFile, double scaleX, double scaleY
-  ) {
+  public BufferedImage scaleImage(File inputFile, double scaleX, double scaleY) {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageEditor.java]: Add JavaDoc

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageHelper.java [hunk 3]
```diff
@@ -26,4 +31,44 @@ public static String getImageId(WebElement image) {
...
+      green.add(pixel.getGreen());
+    }
+    ArrayList<Integer> blue = new ArrayList<>();
+    for (Color pixel : pixels) {
+      blue.add(pixel.getBlue());
+    }
+    return Triple.of(getVariance(red), getVariance(green), getVariance(blue));
+  }
+
+  private static double getVariance(java.util.List<Integer> numbers) {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageHelper.java]: This looks more like average absolute deviation then variance, please rename or change the implementation

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java [hunk 4]
```diff
@@ -111,17 +113,17 @@ public boolean isAdVisible(final WebElement element, final String selector,
       wait.until(new ExpectedCondition<Object>() {
         @Override
         public Object apply(WebDriver driver) {
-          File adImg = shooter.captureWebElement(element, driver);
-          PageObjectLogging.log("ScreenshotsComparison", "Ad image in " + selector, true, driver);
-          boolean areFilesTheSame = imageComparison.areFilesTheSame(backgroundImg, adImg);
-          adImg.delete();
-          return !areFilesTheSame;
+          BufferedImage adImg = shooter.takeScreenshot(element, driver);
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: I strongly recommend you to move calculations to another class, this class is a helper for images on wiki page, not a place to store every image related staff

Comment (drets) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: @ludwikkazmierczak, which calculation an `takeScreenshot` ?
AdsComparison class is exactly for checking such things like whether an ad visible or not.
If you are talking about getting variance, but it's already in the helper images class.

**Comment by gabrys** [comment]
LGTM

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java (hunk 4 above)
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java]: I've missclicked, I'm reffering to ImageHelper and absolute deviation calculation

**Pull request merged**